### ISSUE TITLE
Please release a new version

### ISSUE THREAD

**Issue by thkoch2001** [haskell/haskell-mode#1769]
There have been a few commits since version 17.2 and no commits since September. Would you like to tag current master e.g. as 17.3. so that I also update the Debian package?

**Comment by solna86** [comment]
A new version should be released please. The last tag, 17.2, is almost 3 years old and fails to load on Emacs 29.1.

Therefore, we have to use MELPA instead of MELPA Stable, which is not allowed in some organizations.

**Comment by purcell** [comment]
A bit scrappy, but done: 17.3 is tagged and should build on MELPA Stable shortly.

> which is not allowed in some organizations.

@solna86 very curious to hear about organizations that have policies about which elisp package archives are allowed....

**Comment by solna86** [comment]
Thanks @purcell! I have just tested haskell-mode 17.3 on Emacs 29.1. It loads fine, but there's a repeated error whenever I move the cursor into something:

`eldoc error: (error Eager macro-expansion failure: (error "Misplaced t or ‘otherwise’ clause"))`

This eldoc issue is showing up on other packages when using Emacs 29.1.

> very curious to hear about organizations that have policies about which elisp package archives are allowed....

My corporate IT sadly enforces this policy. We are asked not to use any software repositories labelled as unstable.

**Comment by purcell** [comment]
> eldoc error: (error Eager macro-expansion failure: (error "Misplaced t or ‘otherwise’ clause"))

Hmm, I'll take a look at that if I get a few minutes.

> My corporate IT sadly enforces this policy. We are asked not to use any software repositories labelled as unstable.

MELPA actually isn't labelled unstable. I think of it "snapshot" MELPA, personally. It's a pity about policies like this: usually "unstable" relates to the fact that packaging might be broken at times, but usually banning "unstable" repositories leads to the use of more insecure old packages.

**Comment by djr7C4** [comment]
I had a similar error occur with Emacs 29.1 when requiring 'haskell-mode except that it did not involve eldoc. It was caused by haskell-mode-commands.el. Upgrading to 17.4 fixed the error but it still shows up as a warning in \*Warnings\*.

```
error: Eager macro-expansion failure: (error "Misplaced t or ‘otherwise’ clause")
```

This doesn't seem like a big issue (I don't think it is currently breaking anything for me) but I mention this in case it is useful.

**Comment by purcell** [comment]
It's weird because that error is one produced by `cl-case`, but all the uses of `cl-case` in the haskell-mode files are compliant.

**Comment by djr7C4** [comment]
Some old bytecode may have been loading before or something. I'm no longer getting this error from haskell-mode with 17.4.

**Comment by purcell** [comment]
Yeah, that's the most likely explanation. Personally I do the following to avoid incompatible bytecode resulting from old package installations: https://github.com/purcell/emacs.d/blob/master/lisp/init-elpa.el#L9-L12

**Comment by djr7C4** [comment]
Thanks. I guess maybe the eldoc error others experienced was due to eldoc causing some old bytecode to get autoloaded.
